packages/ckeditor5-ui/src/toolbar/toolbarview.js

```diff
@@ -277,14 +278,24 @@ export default class ToolbarView extends View {
 	 * A utility that expands the plain toolbar configuration into
 	 * {@link module:ui/toolbar/toolbarview~ToolbarView#items} using a given component factory.
 	 *
-	 * @param {Array.<String>} config The toolbar items configuration.
+	 * @param {Array.<String>|Object} itemsOrConfig The toolbar items or the entire toolbar configuration object.
 	 * @param {module:ui/componentfactory~ComponentFactory} factory A factory producing toolbar items.
 	 */
-	fillFromConfig( config, factory ) {
-		this.items.addMany( config.map( name => {
-			if ( name == '|' ) {
+	fillFromConfig( itemsOrConfig, factory ) {
+		const config = normalizeToolbarConfig( itemsOrConfig );
+
+		// Items listed in `config.removeItems` should not be added to the toolbar.
+		const items = config.items.filter( name => config.removeItems.indexOf( name ) === -1 );
+
+		this.items.addMany( items.map( ( name, idx ) => {
```

Comment:
I'd move this `map()` call out of `addMany()`. I thought that `addMany()` accepts a callback at first :D I know it was written this way but it makes sense to change it now.

Also, you can consider using a `reduce()` and not `map()`. Might look better because that empty `return` there is a bit odd now. I actually don't know how it works.

Reply:
Happy to change that.

Reply:
Changed now, but I would discuss this change further.